modules/features2d/src/sift.simd.hpp

```diff
@@ -579,6 +579,11 @@ void calcSIFTDescriptor(
     float *X = buf.data(), *Y = X + len, *Mag = Y, *Ori = Mag + len, *W = Ori + len;
     float *RBin = W + len, *CBin = RBin + len, *hist = CBin + len;
 
+    float* rawDst = 0;
+    cv::utils::BufferArea area;
```

Comment:
It would be better to use the same BufferArea for all temporary locations. Could you please merge it with variables set allocated on `buf`